OrbitGl/TimerTrack.cpp

```diff
@@ -107,14 +107,12 @@ void TimerTrack::UpdatePrimitives(uint64_t min_tick, uint64_t max_tick,
 
         bool is_visible_width = normalized_length * canvas->GetWidth() > 1;
         bool is_selected = &text_box == GOrbitApp->selected_text_box();
-        bool need_highlight = GOrbitApp->selected_text_box() &&
-                              timer_info.thread_id() != SamplingProfiler::kAllThreadsFakeTid &&
-                              timer_info.thread_id() == GOrbitApp->selected_thread_id() &&
-                              timer_info.function_address() ==
-                                  GOrbitApp->selected_text_box()->GetTimerInfo().function_address();
-        need_highlight =
-            need_highlight ||
-            (!GOrbitApp->selected_text_box() && GOrbitApp->highlighted_function() > 0 &&
+        bool need_highlight =
+            (GOrbitApp->selected_text_box() && /*if picking from the capture window*/
+             timer_info.function_address() ==
+                 GOrbitApp->selected_text_box()->GetTimerInfo().function_address()) ||
+            (!GOrbitApp->selected_text_box() && /*if picking from the live functions panel*/
+             GOrbitApp->highlighted_function() > 0 &&
```

Comment:
If you are still leaving the 0, I think that we should have a comment there explaining our assumption that ThreadID are greater than 0. Still, for me is better, if possible, to have "GOrbitApp->highlighted_function() != kFakeTid"

Reply:
Thanks for the suggestion. I chaged to use a constant named kUnusedHighlightedFunctionAddress in the latest fixup commit db98890. But because of my bad naming of those fixup commits, it's hard to notice the purpose of each fixup.